dask/dataframe/optimize.py

```diff
@@ -20,8 +13,11 @@ def optimize(dsk, keys, **kwargs):
     try:
         from castra import Castra
         dsk3 = fuse_getitem(dsk2, Castra.load_partition, 3)
+        def merge(a, b):
+            return (Castra.load_index, b[1], b[2]) if a[2] == 'index' else a
+        dsk4 = fuse_selections(dsk3, getattr, Castra.load_partition, merge)
```

Comment:
> I think I'm still for a more explicit function in dask.dataframe to pull out the index than getattr.

I'm against that simply because I don't think it's needed. The implementation here is simple, and matches only getting an index attribute from a series/dataframe loaded from castra. I can't think of a single situation where this would be incorrect. Also, the similar `getarray` function for dask array results in some import ordering issues to prevent circular imports - the same thing would have to be done here. Until this becomes an issue, I'm -1 on creating a function rather than just using `getattr`.